test/unit-test/dawn/Optimizer/TestsFromSIR/TestStage.cpp

```diff
@@ -0,0 +1,139 @@
+//===--------------------------------------------------------------------------------*- C++ -*-===//
+//                          _
+//                         | |
+//                       __| | __ ___      ___ ___
+//                      / _` |/ _` \ \ /\ / / '_  |
+//                     | (_| | (_| |\ V  V /| | | |
+//                      \__,_|\__,_| \_/\_/ |_| |_| - Compiler Toolchain
+//
+//
+//  This file is distributed under the MIT License (MIT).
+//  See LICENSE.txt for details.
+//
+//===------------------------------------------------------------------------------------------===//
+
+#include "dawn/Compiler/DawnCompiler.h"
+#include "dawn/Compiler/Options.h"
+#include "dawn/SIR/SIR.h"
+#include "dawn/SIR/SIRSerializer.h"
+#include "test/unit-test/dawn/Optimizer/TestEnvironment.h"
+#include <fstream>
+#include <gtest/gtest.h>
+#include <streambuf>
+#include <string>
+
+using namespace dawn;
+
+namespace {
+
+class TestFieldAccessIntervals : public ::testing::Test {
+  std::unique_ptr<dawn::Options> compileOptions_;
+
+  dawn::DawnCompiler compiler_;
+
+protected:
+  TestFieldAccessIntervals() : compiler_(compileOptions_.get()) {}
+  virtual void SetUp() {}
+
+  std::shared_ptr<StencilInstantiation> loadTest(std::string sirFilename) {
+
+    std::string filename = TestEnvironment::path_ + "/" + sirFilename;
+    std::ifstream file(filename);
+    DAWN_ASSERT_MSG((file.good()), std::string("File " + filename + " does not exists").c_str());
+
+    std::string jsonstr((std::istreambuf_iterator<char>(file)), std::istreambuf_iterator<char>());
+
+    std::shared_ptr<SIR> sir =
+        SIRSerializer::deserializeFromString(jsonstr, SIRSerializer::SK_Json);
+
+    std::unique_ptr<OptimizerContext> optimizer = compiler_.runOptimizer(sir);
+    // Report diganostics
+    if(compiler_.getDiagnostics().hasDiags()) {
+      for(const auto& diag : compiler_.getDiagnostics().getQueue())
+        std::cerr << "Compilation Error " << diag->getMessage() << std::endl;
+      throw std::runtime_error("compilation failed");
+    }
+
+    DAWN_ASSERT_MSG((optimizer->getStencilInstantiationMap().count("compute_extent_test_stencil")),
+                    "compute_extent_test_stencil not found in sir");
+
+    return optimizer->getStencilInstantiationMap()["compute_extent_test_stencil"];
+  }
+};
+
+TEST_F(TestFieldAccessIntervals, test_field_access_interval_01) {
+  auto stencilInstantiation = loadTest("test_field_access_interval_01.sir");
```

Comment:
where is this sir file?

Reply:
It is in Optimizer/Passes.... 
I didnt create new sir files, but trying to resue the existing ones

Reply:
ah, ok